relay/sources/rudder-package/src/versions.rs

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2023 Normation SAS
+
+use std::str::FromStr;
+use std::cmp::Ordering;
+use regex::Regex;
+use anyhow::{Context, Ok, Result, bail, Error};
+
+struct ArchiveVersion {
+    pub rudder_version: RudderVersion,
+    pub plugin_version: PluginVersion,
+}
+
+impl FromStr for ArchiveVersion {
+    type Err = Error;
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let re = Regex::new(r"^(?<rudder_version>.*)-(?<plugin_version.*)$")?;
+        let caps = match re.captures(s) {
+            None => bail!("Unparsable rpkg version '{}'", s),
+            Some(c) => c
+        };
+        let rudder_version = RudderVersion::from_str(&caps["rudder_version"].to_string())?;
+        let plugin_version = PluginVersion::from_str(&caps["plugin_version"].to_string())?;
+        Ok(Self {
+            rudder_version,
+            plugin_version
+        })
+    }
+}
+
+ #[derive(PartialEq, Debug)]
+enum RudderVersionMode {
```

Comment:
It looks like Rudder versions used in plugins are simpler than that, without git or nightly versions. We could probably do with:

```rust
    Alpha { version: u16 },
    Beta { version: u16 },
    Rc { version: u16 },
    Final,
```

I'd avoid release as it is also often used in opposition with nightly (and hence used for things like released pre-versions).